code/game/gamemodes/vampire/vampire.dm

```diff
@@ -363,35 +381,46 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 		if(!(n in old_powers))
 			switch(n)
```

Comment:
This switch is pretty bad, can improve it by defining something like this:
```dm
var/list/allPowers[VAMP_MAX] //See below
allPowers[VAMP_SHAPE] =list(
    "gainText" = "You have gained the shapeshifting ability, at the cost of stored blood you can change your form permanently.",
    "verbPath" = /client/proc/vampire_shapeshift)
[...] //Define the rest
```dm
Then that huge switch just turns into:
```dm
if(!(n in old_powers))
    src << allPowers[n]["gainText"]
    verbs += allPowers[n]["verbPath"]
```
A better solution would be to make them datums (the above are faux objects with faux vars).
Then the code would look like:
```dm
/vampower
    var/gainText = "" //Or something basic
    var/verbPath = /client/proc/vampire_error //Or something basic

/vampower/proc/bestow(mob/player/target)
    if(target isn't a vampire) return 0
    if(target isn't powerful enough) return 0
    if(target already has this power) return 0
    target << src.gainText
    target.verbs |= src.verbPath

/vampower/shape
    gainText = "You have gained the shapeshifting ability, at the cost of stored blood you can change your form permanently.",
    verbPath = /client/proc/vampire_shapeshift)

[...] //Define the rest
```
But it looks like it'd take quite a bit of effort to refactor the current code to that.